### ISSUE TITLE
`async_track_state_change` deprecated

### ISSUE THREAD

**Issue by ctih** [claytonjn/hass-circadian_lighting#245]
In Home Assistant new version the function used is deprecated. Could anyone look in to the issue before the release of this release?

Core
2024.5.0b0
Supervisor
2024.04.3
Operating System
12.2
Brugerflade
20240424.1

Logger: homeassistant.helpers.frame
Kilde: helpers/frame.py:190
Første forekomst: 13.34.55 (4 forekomster)
Senest logget: 13.35.01

Detected that custom integration 'circadian_lighting' calls `async_track_state_change` instead of `async_track_state_change_event` which is deprecated and will be removed in Home Assistant 2025.5 at custom_components/circadian_lighting/switch.py, line 234: async_track_state_change(, please create a bug report at https://github.com/claytonjn/hass-circadian_lighting/issues

**Comment by AnthonyBe** [comment]
Yeah, just saw this as well when I upgraded from 2024.4.4 to 2024.5.1
Got a year to fix it, but nice to have one less warning in the log file :)

`2024-05-04 20:33:02.443 WARNING (MainThread) [homeassistant.helpers.frame] Detected that custom integration 'circadian_lighting' calls `async_track_state_change` instead of `async_track_state_change_event` which is deprecated and will be removed in Home Assistant 2025.5 at custom_components/circadian_lighting/switch.py, line 234: async_track_state_change(, please create a bug report at https://github.com/claytonjn/hass-circadian_lighting/issues`

**Comment by angelowolf** [comment]
Hi, I forked the project to fix this as this project seems abandoned. 

The changes are in the switch.py ​​file, you can replace your instance file with this one so it will keep working after the 2024.5 update

**Comment by AnthonyBe** [comment]
I've already moved on from this integration and replaced it with the Adaptive Lighting integration (https://github.com/basnijholt/adaptive-lighting) that was originally inspired by this one, does everything this one does, and more, and is actively maintained.

**Comment by claytonjn** [comment]
For what it's worth, this project is in no way abandoned. This is still the most important component of my personal smart home setup. It works perfectly for me, and the only issues I'm aware of are specific light integrations that do not behave well following the Home Assistant standards and expectations. Almost all of the outstanding feature requests go against the mission of this component. 

I wrote this component for myself and made it available publicly to benefit others if it fits their needs. I'm a full-time software developer with a family and a number of other hobbies, so I don't spend my time adding features or pushing updates out of boredom or just for the sake of GitHub stats. If Circadian Lighting does not align with someone's use-case I absolutely encourage them to use Adaptive Lighting or whatever other integration best fits their needs. That said, I don't want satisfied users to start "jumping ship" just because they think the project is abandoned simply because there haven't been recent commits. A big reason that there aren't a lot of commits is because I intentionally designed CL to be as simple and platform agnostic as possible while providing the features I consider valuable so that it's as stable as possible and works without constant updates.
